### PR TITLE
original Network Admin PR, aka Revert "Revert "Network Admin""

### PR DESCRIPTION
**Pull request by robbiecarlton**
Reverts Hylozoic/hylo-redux#22

### CODE REVIEW

**Comment by robbiecarlton** [comment]
@levity ok, this is ready for you to look at.

File: src/reducers/networks.js [hunk 1]
```diff
@@ -10,17 +13,25 @@ export default function (state = {}, action) {
...
     case CREATE_NETWORK:
       return {
         ...state,
         [payload.slug]: payload
       }
+    case FETCH_COMMUNITIES:
+      let newCommunities = (state[id].communities || []).concat(map(payload.communities, 'id'))
+      return {...state, [id]: {...state[id], communities: newCommunities}}
+    case UPDATE_NETWORK_PENDING:
+      console.log('networks reducer, UNP', {id, params: meta.params})
```
Comment (levity) [inline src/reducers/networks.js]: stray console.log here

**Comment by levity** [comment]
looks good otherwise. it does feel like it's lacking some tests, but i think i need to figure out a way to simulate a client-side environment in the test suite so we can write end-to-end tests where we click buttons and expect state to change accordingly

**Comment by robbiecarlton** [comment]
Agreed, lacking tests, and end to end tests especially. Might be fun to pair on that some...

**Pull request merged**